### PR TITLE
feat!: update poem to v3 & poem-openapi to v5

### PR DESCRIPTION
**Pull request by NexRX**
#### Description:
Current implementation doesn't work with new poem v3 & poem-openapi v5 so I've bumped the versions and made the minor fixes for that migration.

#### Checklist:

- [X] Tests for the changes have been added (_for bug fixes / features_);
   - no tests needed to be updated, all passing locally
- [X] Docs have been added / updated (_for bug fixes / features_).

### CODE REVIEW

File: poem-grants/README.md [hunk 1]
```diff
@@ -118,6 +118,7 @@ async fn manual_secure(details: AuthDetails) -> Response {
 You can find more [`examples`] in the git repository folder and [`documentation`].
 
 ## Supported `poem` versions
+* For `poem-grants: 2.*` supported version of `poem` is `2.*`
```
Comment (DDtKey) [inline poem-grants/README.md:121]: Let's mention 3rd version as well

**Review by DDtKey — APPROVED** [review]
Thank you for the contribution!

File: poem-grants/README.md (hunk 1 above)
Comment (NexRX) [reply poem-grants/README.md:121]: Sure thing! :)

**Comment by NexRX** [comment]
I just noticed the failing tests, not sure why it's working for me but ill fix it.

**Pull request merged**